src/library.cpp

```diff
@@ -444,12 +444,11 @@ is passed to :cpp:func:`lammps_commands_string` for processing.
 
 void lammps_commands_list(void *handle, int ncmd, const char **cmds)
 {
-  LAMMPS *lmp = (LAMMPS *) handle;
   std::string allcmds;
 
   for (int i = 0; i < ncmd; i++) {
     allcmds.append(cmds[i]);
-    if (allcmds.back() != '\n') allcmds.append(1,'\n');
+    if (allcmds.empty() || (allcmds.back() != '\n')) allcmds.append(1,'\n');
   }
```

Comment:
@rbberger somehow this bug went unnoticed by the automated unit tests while it created an exception/segmentation fault on my local machine in the PythonCommands test (you must not use `std::string.back()` on an empty string). Was this just by chance or do some tests fall through the grid of the various configurations?